robot/impl/local_robot.go

```diff
@@ -1106,6 +1106,16 @@ func dialRobotClient(
 // a best effort to remove no longer in use parts, but if it fails to do so, they could
 // possibly leak resources. The given config may be modified by Reconfigure.
 func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config) {
+	r.reconfigure(ctx, newConfig, false)
+}
+
+// ReconfigureSync is like Reconfigure, but prevents any resources from being processed
+// concurrently.
+func (r *localRobot) ReconfigureSync(ctx context.Context, newConfig *config.Config) {
```

Comment:
[q] Why does this need to be exported API? Relatedly, can we just have `Reconfigure` and `reconfigure`?

Reply:
you're right, this test imports `robotimpl` so it will have access to all private methods, and we are not using the sync version of reconfigure anywhere else.